reference/performance.mdx

```diff
@@ -5,36 +5,288 @@ sidebarTitle: Performance
+- **Harder data exploration**: When fields differ drastically between events, discovering the correct columns or shaping queries becomes more difficult.
+
+### How to fix it
+
+- **Keep datasets narrowly focused:** Group data from the same application or service in its own dataset. For example, keep `k8s_logs` separate from `web_traffic`.
+- **Avoid mixing data types for the same field:** Enforce consistent types during ingest. If a field is numeric, always send numeric values.
+- **Consider using map fields:** If you have sparse or high-cardinality nested data, consider storing it in a single map (object) field instead of flattening every key. This reduces the total number of top-level fields. Axiom’s [map fields](/apl/data-types/map-fields#map-fields) are optimized for large objects.
+
+## Excessive backfilling and large `_time` vs. `_sysTime` gaps
+
+### Problem
+
+Axiom’s index is primarily built around time. Ideally, incoming events for a block lie in a closely bounded time range. However, backfilling large amounts of historical data after the fact (especially out of chronological order) creates wide time overlaps in blocks. If `_time` is far from `_sysTime` (the time the event was ingested), Axiom’s primary time index is weakened.
+
+### Why it matters
+
+- **Poor performance on time-based queries**: Blocks must be scanned despite time filters, because many blocks overlap the query time window.
+- **Inefficient block filtering**: Queries that filter on time must scan blocks that contain data from many time segments.
+- **Large data merges**: Compaction processes that rely on time ordering become less efficient.
+
+### How to fix it
+
+- **Minimize backfill:** Try to ingest events close to their actual creation time whenever possible. Ingest events close to the time they occur.
+- **Backfill in dedicated batches:** If you must backfill older data, do it in dedicated batches that do not mix with live data.
+- **Use discrete backfill intervals:** When backfilling data, ingest one segment at a time (for example, day-by-day).
+- **Avoid wide time ranges in a single batch:** If you are sending data for a 24-hour period, avoid mixing in data that is weeks or months older.
+- **Be aware of ingestion concurrency:** Avoid mixing brand-new events with extremely old events in the same ingest request.
+
+<Note>
+Future improvements: Axiom’s roadmap includes an initiative which aims to mitigate the impact of poorly clustered time data by performing incremental time-based compaction. Until then, avoid mixing large historical ranges with live ingest whenever possible.
+</Note>
+
+## Large number of fields in a dataset
+
+### Problem
+
+Slow query performance in datasets with very high dimensionality (with more than several thousand fields).
+
+### Why it matters
 
 Axiom stores event data in a tuned format. As a result:
 - The number of distinct values (cardinality) in your data has little impact on performance.
```

Comment:
We may not want to make this point here, but technically, low cardinality fields compress better than high cardinality fields, so it does impact performance in the sense that we achieve higher compression ratios than otherwise.